wtxmgr/unconfirmed.go

```diff
@@ -0,0 +1,172 @@
+		prevOutKey := canonicalOutPoint(&prevOut.Hash, prevOut.Index)
+		doubleSpendHash := existsRawUnminedInput(ns, prevOutKey)
+		if doubleSpendHash != nil {
+			var doubleSpend TxRecord
+			doubleSpendVal := existsRawUnmined(ns, doubleSpendHash)
+			copy(doubleSpend.Hash[:], doubleSpendHash) // Silly but need an array
+			err := readRawTxRecord(&doubleSpend.Hash, doubleSpendVal,
+				&doubleSpend)
+			if err != nil {
+				return err
+			}
+
+			log.Debugf("Removing double spending transaction %v",
+				doubleSpend.Hash)
+			err = s.removeConflict(ns, &doubleSpend)
+			if err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+// removeConflict removes an unmined transaction record and all spend chains
+// deriving from it from the store.  This is designed to remove transactions
+// that would otherwise result in double spend conflicts if left in the store,
+// and to remove transactions that spend coinbase transactions on reorgs.
+func (s *Store) removeConflict(ns walletdb.Bucket, rec *TxRecord) error {
+	// For each potential credit for this record, each spender (if any) must
+	// be recursively removed as well.  Once the spenders are removed, the
+	// credit is deleted.
+	numOuts := uint32(len(rec.MsgTx.TxOut))
+	for i := uint32(0); i < numOuts; i++ {
+		k := canonicalOutPoint(&rec.Hash, i)
+		spenderHash := existsRawUnminedInput(ns, k)
+		if spenderHash != nil {
+			var spender TxRecord
+			spenderVal := existsRawUnmined(ns, spenderHash)
+			copy(spender.Hash[:], spenderHash) // Silly but need an array
+			err := readRawTxRecord(&spender.Hash, spenderVal, &spender)
+			if err != nil {
+				return err
+			}
+
+			log.Debugf("Transaction %v is part of a removed conflict "+
+				"chain -- removing as well", spender.Hash)
+			err = s.removeConflict(ns, &spender)
+			if err != nil {
+				return err
+			}
+		}
+		err := deleteRawUnminedCredit(ns, k)
+		if err != nil {
+			return err
+		}
+	}
+
+	// If this tx spends any previous credits (either mined or unmined), set
+	// each unspent.  Mined transactions are only marked spent by having the
+	// output in the unmined inputs bucket.
+	for _, input := range rec.MsgTx.TxIn {
+		prevOut := &input.PreviousOutPoint
+		k := canonicalOutPoint(&prevOut.Hash, prevOut.Index)
+		err := deleteRawUnminedInput(ns, k)
+		if err != nil {
+			return err
+		}
+	}
+
+	return deleteRawUnmined(ns, rec.Hash[:])
+}
+
+// UnminedTxs returns the underlying transactions for all unmined transactions
+// which are not known to have been mined in a block.
+func (s *Store) UnminedTxs() ([]*wire.MsgTx, error) {
+	var txs []*wire.MsgTx
+	err := scopedView(s.namespace, func(ns walletdb.Bucket) error {
+		var err error
+		txs, err = s.unminedTxs(ns)
+		return err
+	})
+	return txs, err
+}
+
+func (s *Store) unminedTxs(ns walletdb.Bucket) ([]*wire.MsgTx, error) {
+	var unmined []*wire.MsgTx
+	err := ns.Bucket(bucketUnmined).ForEach(func(k, v []byte) error {
+		// TODO: Parsing transactions from the db may be a little
```

Comment:
There aren't any callers that would be able to use this.  `btcrpcclient` forces transactions to be passed as a `*wire.MsgTx`.